text/0072-kafka-schema-registry.md

```diff
@@ -0,0 +1,111 @@
+- Start Date: 2023-02-01
+- RFC Type: feature
+- RFC PR: n/a
+- RFC Status: draft
+
+# Summary
+
+This RFC proposes introducing a centralized schema repository for the Kafka topics in use at Sentry. It provides mappings between the Kafka topics we are using to the encoding format and schemas for messages on that topic.
+
+Topic registration and schema validation of messages will be optional for the foreseeable future.
+
+# Motivation
+
+Kafka is increasingly used as the message bus between services at Sentry. Kafka topics and their schemas are usually not internal to any one service, but part of the contract between services.
+
+As Sentry grows, the number of message types and topics in use is exploding. We are marching further towards increasingly distributed ownership of data by product and infrastructure engineering teams. We have already seen an increase in the number of incidents related with invalid data and schema issues at Sentry. It is reasonable to expect this trend to continue as the number of topics, data types, teams and engineers contributing to Sentry grows.
+
+The goals of the centralized schema repository are to:
```

Comment:
One thing which I have found useful with other schema repositories in the past is having example messages for each schema type. Looking at schemas versus actual example messages complying with the schema are two different things. Once example messages are there for schemas its easy for consumers to write tests using example messages. Should we consider adding an example message for each schema in the schema repository a requirement to make it easier for consumers to write tests?

Reply:
Added to the list of goals